Rules/UseConsistentCasing.cs

```diff
@@ -0,0 +1,235 @@
+using Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic;
+using System.Management.Automation;
+using System.Linq;
+#if !CORECLR
+using System.ComponentModel.Composition;
+#endif
+using System.Globalization;
+
+namespace Microsoft.Windows.PowerShell.ScriptAnalyzer.BuiltinRules
+{
+    /// <summary>
+    /// UseConsistentCasing: Check if cmdlet is cased correctly.
+    /// </summary>
+#if !CORECLR
+    [Export(typeof(IScriptRule))]
+#endif
+
+    public class UseConsistentCasing : ConfigurableRule
+    {
+        /// <summary>If true, require the case of all operators to match the specified casing.</summary>
+        [ConfigurableRuleProperty(defaultValue: "true")]
+        public bool CheckOperator { get; set; }
+
+        /// <summary>If true, require the case of all keywords to match the specified casing.</summary>
+        [ConfigurableRuleProperty(defaultValue: "true")]
+        public bool CheckKeyword { get; set; }
+
+        /*
+         * Hypothetically, we could support UPPERCASE but that is awful, and ...
+         * Anything else would require this analyzer to **make up** the correct capitalization.
+        private enum Casing { uppercase, lowercase }
+        private Casing casing;
+        /// <summary>By default, switch to lowercase (or uppercase (defaults to lowercase)</summary>
+        /// <remarks>Any other "consistent" case would just </remarks>
+        [ConfigurableRuleProperty(defaultValue: "lowercase")]
+        public string Case
+        {
+            get
+            {
+                return casing.ToString();
+            }
+            set
+            {
+                if (String.IsNullOrWhiteSpace(value) || !Enum.TryParse<Casing>(value, true, out casing))
+                {
+                    casing = Casing.lowercase;
+                }
+            }
+        }
+        */
+
+        /// <summary>
+        /// AnalyzeScript: Analyze the script to verify consistency of keyword and operator case
+        /// </summary>
+        public override IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
+        {
+            if (ast == null) throw new ArgumentNullException(Strings.NullAstErrorMessage);
+
+            Token[] tokens;
+            ParseError[] errors;
+
+            if (CheckKeyword) {
+                // sorry, but we had to re-parse, because the AST doesn't give us keywords...
+                Parser.ParseInput(ast.Extent.Text, out tokens, out errors);
+
+                // Iterates all keywords and check the case
+                foreach (Token keyword in tokens.Where(t => t.TokenFlags.HasFlag(TokenFlags.Keyword)))
+                {
+                    string correctCase = keyword.Text.ToLowerInvariant();
+                    if (!keyword.Text.Equals(correctCase, StringComparison.Ordinal))
+                    {
+                        yield return GetDiagnosticRecord(keyword, fileName, correctCase);
+                    }
+                }
```

Comment:
Eliminates a few allocations like this (especially the `Enum.HasFlag()` method, which has to do a bunch of boxing in .NET Framework):

```suggestion
                for (int i = 0; i < Helper.Tokens.Length; i++)
                {
                    Token currToken = Helper.Tokens[i];
                    
                    if ((currToken.TokenFlags & TokenFlags.Keyword) == 0)
                    {
                        continue;
                    }
                    
                    string correctCase = currToken.Text.ToLowerInvariant();
                    if (!currToken.Text.Equals(correctCase, StringComparison.Ordinal))
                    {
                        yield return GetDiagnosticRecord(keyword, fileName, correctCase);
                    }
                }
```